### PR TITLE
refactor: address React v15.x warnings, while remaining compat

### PR DESCRIPTION
**Pull request by jsg2021**
resolves warnings when the proxy component is defined/loaded on React 15.5.x

### CODE REVIEW

File: mixinReactProxy.js [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+
+function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
+
+function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }
+
+function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }
```
Comment (d3viant0ne) [inline mixinReactProxy.js]: Why are we pulling helper functions directly out of babel into src? The legal issues alone prevent this from being merged, this problem is going to have to be solved another way.

**Comment by jsg2021** [comment]
My bad, I didn't even think of that. I've rewritten to simply create a de-sugared class (function prototype following the es6 class definition rules.)

**Comment by isunnyvirk** [comment]
Hi are there any updates to this PR, or if theres anything I can do to get this out?

**Comment by jsg2021** [comment]
I've been running on this branch for several months... though, I'm contemplating moving to a more explicit pattern with [react-loadable](https://github.com/thejameskyle/react-loadable)

**Pull request closed without merging**